design/open/lakefs-kv-execution-plan.md

```diff
@@ -0,0 +1,72 @@
+# lakeFS on KV - Execution Plan
+
+Key/value data storage package (kv) that will supply key/value access to data to replace the current storage and locking done by Postgres.
+High level overview of the interface described in https://github.com/treeverse/lakeFS/blob/ed55edc2c24b24ed4e6fcccad1ae95844d38b9ad/design/open/metadata_kv/index.md. The package will support multiple back-end implementations.
+
+## Transform from using db to kv package
+
+High level interface described in the kv proposal, we should consider taking the DynamoDB interface to address the way we communicate `set-if` functionality while marshal the same value without fetching data first.
+
+Key/value format can be done using ProtoBuf / JSON.
+
+- key format based on path - a format with escaping should be supported to enable easy way to encode values as part of the key
+- each model should include version as part of the model information
+- migrate from old model to new one without downtime
```

Comment:
Is it the same model as the previous line? Or are we talking about current implementation vs. new kv store implementation?

Reply:
Removed the use of the word data. But just to be clear, yes - each data we keep in a table and would be modeled into the key/value the above text consider as 'model'.